### PR TITLE
support IPNS .eth domains

### PR DESCRIPTION
**Pull request by Leibniz137**

### CODE REVIEW

**Comment by dapplion** [comment]
@Leibniz137 This looks great! Thank you so much for the contribution. Feel free to mark as "Read for review" when appropiate.

@pablomendezroyo Can you test this branch on a DAppNode and attempt to resolve `uniswap.eth`? Maybe we should add a unit test for it

**Comment by dapplion** [comment]
@pablomendezroyo Bump

**Review by dapplion — DISMISSED** [review]

**Comment by dapplion** [comment]
@Leibniz137 I've been testing this approach and I can resolve uniswap.eth to `ipns-ns/12uA8M8Ku8mHUumxHcu7uee` but the IPFS gateway doesn't know what to do with it. Have you been able to resolve that result?

File: packages/dappmanager/test/ethForward/resolveDomain.test.int.ts [hunk 1]
```diff
@@ -48,6 +55,14 @@ describe("ethForward > resolveDomain", () => {
     }
   });
 
+  describe("Fetch IPNS content", () => {
+    it.skip("Should resolve IPNS content", async () => {
+      // Resolve uniswap.eth
+      const res = await fetch(`${ipfsGateway}/ipns/12uA8M8Ku8mHUumxHcu7uee`);
```
Comment (Leibniz137) [inline packages/dappmanager/test/ethForward/resolveDomain.test.int.ts:61]: I just do `/ipns/uniswap.eth` (ie pass the name not the hash)

File: packages/dappmanager/src/ethForward/ipfsProxy.ts [hunk 2]
```diff
@@ -88,18 +88,14 @@ export function getIpfsProxyHandler<T>(
...
-): string {
+function getTargetUrl(proxyType: ProxyType, content: Content): string {
   switch (proxyType) {
     case ProxyType.ETHFORWARD:
       switch (content.location) {
         case "ipfs":
           return urlJoin(params.ETHFORWARD_IPFS_REDIRECT, "ipfs", content.hash);
         case "ipns":
-          return urlJoin(params.ETHFORWARD_IPFS_REDIRECT, "ipns", domain);
+          return urlJoin(params.ETHFORWARD_IPFS_REDIRECT, "ipns", content.hash);
```
Comment (Leibniz137) [inline packages/dappmanager/src/ethForward/ipfsProxy.ts:98]: here i think you actually want to keep passing the domain, we don't need to resolve it to the hash

**Pull request closed without merging**